docs/_dg/TaggingFunctionality.md

```diff
@@ -0,0 +1,47 @@
+<!-- markdownlint-disable-file first-line-h1 -->
+### Tagging functionality
```

Comment:
Yes I think this is good, especially in explaining the broader concept of our tag system. We can put it under the class implementation details (before the other command details). Since I don't see that you have included this anywhere in the DG/Implementation file, can I check where you plan to put it in?